src/Spago/Build.hs

```diff
@@ -242,16 +242,18 @@ repl newPackages sourcePaths pursArgs depsOnly = do
 --   (or the provided module name) with node
 test :: HasBuildEnv env => Maybe ModuleName -> [BackendArg] -> RIO env ()
 test maybeModuleName extraArgs = do
+  logDebug "Running `Spago.Build.test`"
   let moduleName = fromMaybe (ModuleName "Test.Main") maybeModuleName
   Config.Config { alternateBackend } <- view (the @Config)
-  maybeGraph <- view (the @Graph)
+
   -- We check if the test module is included in the build and spit out a nice error if it isn't (see #383)
-  for_ maybeGraph $ \(ModuleGraph moduleMap) -> case Map.lookup moduleName moduleMap of
-    Nothing -> die [ "Module '" <> (display . unModuleName) moduleName <> "' not found! Are you including it in your build?" ]
-    Just _ -> do
-        sourceDir <- Turtle.pwd
-        let dirs = RunDirectories sourceDir sourceDir
-        runBackend alternateBackend dirs moduleName (Just "Tests succeeded.") "Tests failed: " extraArgs
+  maybeGraph <- view (the @Graph)
+  for_ maybeGraph $ \(ModuleGraph moduleMap) -> when (isNothing $ Map.lookup moduleName moduleMap) $
```

Comment:
Hrm, since the graph is only supported in `0.14` and up, I think we run into the risk of `moduleName` not existing, but us not catching it up front for anything below 0.14.

Reply:
That's correct. I think it's a fine tradeoff: people will get the check for `purs` version older than `0.14.0` and any `spago` version, or for `spago` versions earlier than `0.20` and any `purs` version.

The demographic exposed to the "confusing error when the test module doesn't exist" is only composed by people like the bug reporter, that presumably have an existing build and are just upgrading `spago`, but do not want to yet upgrade `purs`. Since one runs into these errors only when setting up a build and getting the module name wrong, I believe the inconvenience for users is going to be minimal as this is not going to show up in practice.

We used to have a module name parser to provide this check before the graph command, and I'd not like to reintroduce that

Reply:
Okay! Yeah, the tradeoff seems fine. I agree that avoiding adding the module name parser back in is very preferable :D